openghg/store/base/_datasource.py

```diff
@@ -577,8 +579,60 @@ def save(self, bucket: str) -> None:
                 if not parent_folder.exists():
                     parent_folder.mkdir(parents=True, exist_ok=True)
 
-                data.to_netcdf(filepath, engine="netcdf4")
+                if compression:
+                    # variables with variable length data types shouldn't be compressed
+                    # e.g. object ("O") or unicode ("U") type
+                    do_not_compress = []
+
+                    # regex for Unicode and Object dtypes, with character code U or O
+                    # type strings may start with a byteorder character: <, >, =, or |,
+                    # so we skip these if present.
+                    dtype_pat = re.compile(r"[<>=|]?[UO]")
+                    for dv in data.data_vars:
+                        if dtype_pat.match(data[dv].data.dtype.str):
+                            do_not_compress.append(dv)
+
+                    # setting compression levels for data vars in data
+                    comp = dict(zlib=True, complevel=5)
+
+                    valid_encoding_keys = [
+                        "zlib",
+                        "fletcher32",
+                        "dtype",
+                        "complevel",
+                        "chunksizes",
+                        "compression",
+                        "shuffle",
+                        "contiguous",
+                        "least_significant_digit",
+                        "_FillValue",
+                    ]
+                    encoding = {}
+                    for var in data.data_vars:
+                        if var not in do_not_compress:
+                            enc = {k: v for k, v in data[var].encoding.items() if k in valid_encoding_keys}
+                            enc.update(comp)
+                            encoding[var] = enc
 
+                    try:
+                        data.to_netcdf(filepath, engine="netcdf4", encoding=encoding)
+                    except RuntimeError:
+                        logger.warning(
+                            "Storing footprint for date range {daterange} without compression due to netCDF4 RuntimeError."
```

Comment:
This needs to be an f-string

Reply:
I'm getting some of these RuntimeErrors when I run the tests, can we have a look at what's causing them?

Reply:
I tried tracing what goes wrong. The variable that causes the problem is `rn_number_of_observations` for the WAO data in one of the fixtures in tests/analyse/conftest.py. This variable doesn't have chunking set... I'm not sure why it raises an issue. The error comes from the netcdf-c code base, and it's a pain to try to find where things are actually defined. For some reason the error comes from a function that has to do with chunking, but there is no chunking... so I have no idea what is wrong.